mwdb/web/src/components/Settings/Views/OAuthRegister.jsx

```diff
@@ -47,9 +69,81 @@ export default function OAuthRegister() {
         }
     }
 
+    async function updateDiscoverData() {
+        try {
+            var req = new Request(discoverURL);
+            fetch(req)
+                .then(async (response) => {
+                    return await response.json();
+                })
+                .then((x) => {
+                    setDiscoverData(x);
+                });
+        } catch (e) {
+            viewAlert.setAlert({ e });
+        }
+    }
+
+    const getDiscover = useCallback(updateDiscoverData, [
+        discoverURL,
+        viewAlert,
+    ]);
+    // eslint-disable-next-line
+    const autoFillCallback = useCallback(autoFill, [discoverData]);
+
+    useEffect(() => {
+        getDiscover();
+    }, [getDiscover]);
+    useEffect(() => {
+        autoFillCallback();
+    }, [autoFillCallback]);
```

Comment:
Client-side approach violates Same Origin Policy, this can't work without special Access-Control-Allow-Origin headers on OIDC Provider side which is not guaranteed.

Reply:
I would move that part to separate PR if possible